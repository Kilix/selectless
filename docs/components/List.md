# List

The default component is a `<div>`.

| property     | type/description                                                                           |
|--------------|--------------------------------------------------------------------------------------------|
| `setRef`     | function used for React reference                                                          |
| `renderItem` | the React Component used to render the items list                                          |
| `render`     | `({opened: boolean, items: Array[<ReactElement<*>], setRef: function}) => ReactElement<*>` |


# Item

The default component is a `<div>` with an `onClick` bind to `onSelectValue`.

| property     | type/description                                                                                        |
|--------------|---------------------------------------------------------------------------------------------------------|
| `currentRef` | function use for React reference                                                                        |
| `render`     | `({data: Option, isCurrent: boolean, isSelected: boolean, onSelectValue: function}) => ReactElement<*>` |