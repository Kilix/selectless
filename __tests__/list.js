import React from 'react'
import renderer from 'react-test-renderer'

import Select from '../src/'
import List from '../src/components/list'
import Item from '../src/components/item'

const options = [{label: 'Paris', value: 'paris'}, {label: 'Tokyo', value: 'tokyo'}]
it('List', () => {
  const tree = renderer
    .create(
      <Select name="list" options={options} defaultValue={options[0]}>
        <List renderItem={Item} />
      </Select>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
it('List: Custom Item', () => {
  const Ctem = () => <div />
  const tree = renderer
    .create(
      <Select name="list" options={options} defaultValue={options[0]}>
        <List renderItem={<Item render={Ctem} />} />
      </Select>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
