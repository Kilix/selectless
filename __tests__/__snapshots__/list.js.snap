// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List 1`] = `
<div>
  <input
    name="list[Paris]"
    type="hidden"
    value="paris"
  />
</div>
`;

exports[`List: Custom Item 1`] = `
<div>
  <input
    name="list[Paris]"
    type="hidden"
    value="paris"
  />
</div>
`;
